cmd/agent/subcommands/flare/command_test.go

```diff
@@ -86,6 +94,8 @@ func TestCommandTestSuite(t *testing.T) {
 
 func (c *commandTestSuite) TestReadProfileData() {
 	t := c.T()
+	RestartSystemProbeTestServer(c)
```

Comment:
I'd suggest starting the server in tests that need it, and shutting it down with `t.Cleanup`. That would get rid of the need for these restart calls. Generally, leaving things running after a test completes is bad practice.

Reply:
Good call.